src/test/java/GameLoopTest.java

```diff
@@ -0,0 +1,55 @@
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class GameLoopTest {
```

Comment:
Nice isolated test. You may find that you want to extract your fakes at some point to reuse them across test classes, but this is good for now.